packages/typespec-python/src/emitter.ts

```diff
@@ -229,11 +241,10 @@ function getBodyType(context: SdkContext, route: HttpOperation): Record<string,
             }
         }
         if (resourceType && bodyModel.name === "") {
-            const effectivePayloadType = getEffectivePayloadType(context, bodyModel);
-            if (effectivePayloadType.name !== "") {
-                bodyModel = effectivePayloadType;
-            } else {
+            if (isModelReferredInTemplate(bodyModel, resourceType)) {
```

Comment:
User may define anonymous model on purpose so we can't use name as judgement. The better way is to see whether the resourceType is referred in template.
Thanks [hint](https://github.com/Azure/autorest.java/blob/9d4df9fc332b4379a112ebe9826c9da8b4b98912/typespec-extension/src/code-model-builder.ts#L760-L772) from @weidongxu-microsoft

Reply:
I don't quite get the idea. For the tsp in the issue, the body should be `AddFaceFromUrlContent`, right?

Reply:
Yep.